helix-term/src/commands.rs

```diff
@@ -916,6 +918,34 @@ fn extend_char_right(cx: &mut Context) {
     doc.set_selection(view.id, selection);
 }
 
+fn extend_char_up(cx: &mut Context) {
+    let count = cx.count();
+    let (view, doc) = current!(cx.editor);
+    let text = doc.text().slice(..);
+    let old_selection = doc.selection(view.id).clone();
+    let mut ranges: SmallVec<_> = old_selection.ranges().into();
+    for r in old_selection.iter() {
+        ranges.push(Range::point(
+            movement::move_vertically(text, *r, Direction::Backward, count, Movement::Extend).head,
```

Comment:
Okay one last thing: The count should duplicate the selection on `n` lines, for example `3C` should duplicate it on all three lines above the current line. `move_vertically`'s count will instead give you a single range that's `n` lines above.

Reply:
Ah good point; I forgot about that...